src/utils.js

```diff
@@ -0,0 +1,199 @@
+};
+
+const processArray = function (arrayProp) {
+  if (VERBOSE) console.log(`Processing array ${JSON.stringify(arrayProp[0])}`);
+  const items = arrayProp[1].items;
+  if (items["$ref"]) {
+    if (VERBOSE) console.log(`Found multiple references ${items["$ref"]}`);
+    return [processAssociation(arrayProp), "objects"];
+  }
+  else if (items["anyOf"]) {
+    if (VERBOSE) console.log(`Found multiple $ref ${items.items}`);
+    let outputRelations = []
+    items['anyOf'].forEach((item) => {
+      outputRelations.push(
+          {
+            name: `${arrayProp[0]}_${getSchemaName(item["$ref"])}`,
+            value: processAnyOf({"anyOf": [item]})
+          }
+      )
+    })
+    return [outputRelations, "multiples"]
+  }
+  else if (!items.type) {
+    if (VERBOSE) console.log(`Found multiple scalars ${items.type}`);
+    return [`[${recognizeIsaType(arrayProp)}]`, "scalars"];
+  }
+  return null;
+};
+
+const processAssociation = function (assocProp) {
+  if (VERBOSE) console.log(`Processing association ${JSON.stringify(assocProp[0])}`);
+  let to_many = "items" in assocProp[1],
+    references = "items" in assocProp[1] ? assocProp[1].items : assocProp[1],
+    schemaName = "";
+
+  if (Object.keys(references).includes("$ref")) {
+    // there is a single reference to another schema
+    schemaName = getSchemaName(references["$ref"]);
+    schemaName in relationMapping
+      ? relationMapping[schemaName].push(PARENT)
+      : (relationMapping[schemaName] = [PARENT]);
+    return [
+      { [assocProp[0]]: relationTemplate(to_many, schemaName, "To-Do-Key", "To-Do-keyIn") },
+      assocProp[0] + "_fk",
+    ];
+  }
+
+  else if (Object.keys(references).includes("anyOf")) {
+    const anyOf = processAnyOf(assocProp[1], to_many);
+    if (anyOf === 'String') return [null, null]
+    else {
+      return [
+        { [assocProp[0]]: anyOf },
+        assocProp[0] + "_fk",
+      ];
+    }
+  }
+  return [null, null]
+};
+
+function processAnyOf(fieldProp, to_many) {
+  if (VERBOSE)  console.log(`Processing anyOf ${JSON.stringify(fieldProp)}`);
+  const item = fieldProp['anyOf'].filter(element => "$ref" in element)[0];
+  if (item) {
+    const target = getSchemaName(item["$ref"]),
+        keyIn = to_many ? PARENT : target,
+        targetKey = `${target}_id`;
+    target in relationMapping
+        ? relationMapping[target].push(PARENT)
+        : (relationMapping[target] = [PARENT]);
+    return relationTemplate(to_many, target, targetKey, keyIn);
+  }
+  return "String"
+}
+
+const processScalar = function (scalarProp) {
+  if (VERBOSE) console.log(`Processing scalar ${scalarProp[0]}`);
+  if (scalarProp[0] !== "@id") {
+    const result = { [scalarProp[0]]: recognizeIsaType(scalarProp[1]) };
+    if (VERBOSE) console.log(result);
+    return result;
+  }
+};
+
+function getSchemaName(string) {
+  return string.replace("_schema.json", "").replace("#", "");
+}
+
+function relationTemplate(toMany, schemaName, targetKey, keysIn) {
```

Comment:
something to check: is it "keysIn" or "keyIn"?

Reply:
Its always `keysIn`, independent of the type of the association. The old incomplete prototype Zendro "ISA-models" have some outdated attributes. We changed `keyIn` to `keysIn` at some point.